tests/manage/cluster/nodes/test_nodes_restart.py

```diff
@@ -0,0 +1,206 @@
+import logging
+import pytest
+
+from ocs_ci.framework import config
+from ocs_ci.utility.utils import ceph_health_check
+from ocs_ci.ocs import ocp, constants
+from ocs_ci.framework.testlib import tier4, ignore_leftovers, ManageTest
+from ocs_ci.utility import aws
+from ocs_ci.ocs.cluster import CephCluster
+from tests import helpers
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def aws_obj():
+    """
+    Returns:
+        AWS: An instance of AWS class
+
+    """
+    aws_obj = aws.AWS()
+    return aws_obj
+
+
+@pytest.fixture()
+def instances(request, aws_obj):
+    """
+    Get cluster instances
+
+    Returns:
+        dict: The ID keys and the name values of the instances
+
+    """
+    nodes = ocp.get_all_nodes()
+    ec2_instances = aws.get_instances_ids_and_names(nodes)
+
+    def finalizer():
+        """
+        Make sure all instances are running
+        """
+        stopping_instances = {
+            key: val for key, val in ec2_instances.items() if (
```

Comment:
If we're not using the values, I think it's better to iterate only over the dictionary's keys
```suggestion
            instanceid for instanceid in ec2_instances if (
```